src/_components/hadith-data/hadith-data.js

```diff
@@ -0,0 +1,118 @@
+import moment from 'moment';
+import config from '../../config.json';
+import csvtojson from 'csvtojson';
+import request from 'request';
+
+export default class HadithData {
+  constructor() {
+    this.getHadithFromGoogleSheets().then(() => {
```

Comment:
This causes the app to request the data from GoogleSheets to be called every time the view is show. This should be only done based on the refresh period set.

Reply:
Ah, yh it does, was meant to call the update function in the constructor, I'll change this.